File: backend/src/dto/person_detail.rs
use serde::{Deserialize, Serialize};

/// Response for GET /persons/:id/detail
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct PersonDetailResponse {
    pub person: PersonInfo,
    pub summary: PersonSummary,
    pub documents: Vec<DocumentGroup>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct PersonInfo {
    pub id: String,
    pub name: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub role: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct PersonSummary {
    pub total_statements: i64,
    pub documents_count: i64,
    pub characterizations_count: i64,
    pub rebuttals_received_count: i64,
}

/// Statements grouped by source document
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DocumentGroup {
    pub document_id: String,
    pub document_title: String,
    pub statement_count: usize,
    pub statements: Vec<StatementDetail>,
}

/// A single evidence statement with optional characterization and rebuttal info
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct StatementDetail {
    pub evidence_id: String,
    pub title: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub verbatim_quote: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub page_number: Option<i64>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub kind: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub significance: Option<String>,
    pub characterizes: Vec<CharacterizesInfo>,
    pub rebutted_by: Vec<RebuttalInfo>,
}

/// Which allegation a statement characterizes and how
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CharacterizesInfo {
    pub allegation_id: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub allegation_text: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub characterization_label: Option<String>,
}

/// Evidence that rebuts a statement
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct RebuttalInfo {
    pub evidence_id: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub title: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub verbatim_quote: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub stated_by: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub document_title: Option<String>,
}

File: frontend/src/services/personDetail.ts
import { API_BASE_URL } from "./api";
import { authFetch } from "./auth";

export type PersonInfo = {
  id: string;
  name: string;
  role?: string;
};

export type PersonSummary = {
  total_statements: number;
  documents_count: number;
  characterizations_count: number;
  rebuttals_received_count: number;
};

export type CharacterizesInfo = {
  allegation_id: string;
  allegation_text?: string;
  characterization_label?: string;
};

export type RebuttalInfo = {
  evidence_id: string;
  title?: string;
  verbatim_quote?: string;
  stated_by?: string;
  document_title?: string;
};

export type StatementDetail = {
  evidence_id: string;
  title: string;
  verbatim_quote?: string;
  page_number?: number;
  kind?: string;
  significance?: string;
  characterizes: CharacterizesInfo[];
  rebutted_by: RebuttalInfo[];
};

export type DocumentGroup = {
  document_id: string;
  document_title: string;
  statement_count: number;
  statements: StatementDetail[];
};

export type PersonDetailResponse = {
  person: PersonInfo;
  summary: PersonSummary;
  documents: DocumentGroup[];
};

export async function getPersonDetail(
  personId: string,
): Promise<PersonDetailResponse> {
  const response = await authFetch(
    `${API_BASE_URL}/persons/${encodeURIComponent(personId)}/detail`,
  );

  if (!response.ok) {
    if (response.status === 404) {
      throw new Error("Person not found");
    }
    throw new Error(`Failed to fetch person detail: ${response.status}`);
  }

  return response.json();
}
